### ISSUE TITLE
Implement cookie notice and selection into Mastodon

### ISSUE THREAD

**Issue by br4yd** [mastodon/mastodon#21396]
### Pitch

Having the option to turn on a cookie banner (optionally with selection option) in the settings would be great. Especially when running a Mastodon instance in the EU you technically are forced to implement a cookie notice on the site and let users select which type of cookies they want to use.

### Motivation

- Regulations inside the EU
- Adding a cookie banner to Mastodon without being a developer is very difficult

**Comment by rbairwell** [comment]
Since the main developer of Mastodon (Gargron) is in Germany, I do expect them to be already aware of these regulations.

However, Mastodon only sets essential/functional cookies needed for the site to operate (such as holding your login state)  - no third party cookies, no advertising cookies etc so it is my belief that it is exempt.

**Comment by madbeardman** [comment]
> Since the main developer of Mastodon (Gargron) is in Germany, I do expect them to be already aware of these regulations.
> 
> However, Mastodon only sets essential/functional cookies needed for the site to operate (such as holding your login state) - no third party cookies, no advertising cookies etc so it is my belief that it is exempt.

Correct, as no 3rd party cookies are required (for tracking etc) there's no need to show a banner and ask for consent. I would suggest a banner though saying there are no tracking cookies/3rd party cookies, as according to testing we do in public websites, people are seeing the banner as a thing to look for and trust.

**Comment by br4yd** [comment]
Mastodon stores IPs longer than needed for e.g. let alone therefore you need a cookie policy in the EU afaik. I'm not a lawyer but I'd rather prefer the possibility of adding one and never needing it over not being able to implement one when it's needed.
